File: elevaite/python_apps/agent_studio/agent-studio/db/schemas/agents.py

import uuid
from datetime import datetime
from typing import Dict, List, Optional, Literal, Any
from openai.types.chat.chat_completion_tool_param import ChatCompletionToolParam

from pydantic import BaseModel, ConfigDict

from .prompts import PromptResponse

class AgentBase(BaseModel):
    name: str
    agent_type: Optional[
        Literal[
            "router",
            "web_search",
            "data",
            "troubleshooting",
            "api",
            "weather",
            "toshiba",
        ]
    ] = None
    description: Optional[str] = None
    parent_agent_id: Optional[uuid.UUID] = None
    system_prompt_id: uuid.UUID
    persona: Optional[str] = None
    routing_options: Dict[str, str]
    short_term_memory: bool = False
    long_term_memory: bool = False
    reasoning: bool = False
    input_type: List[Literal["text", "voice", "image"]] = ["text", "voice"]
    output_type: List[Literal["text", "voice", "image"]] = ["text", "voice"]
    response_type: Literal["json", "yaml", "markdown", "HTML", "None"] = "json"
    max_retries: int = 3
    timeout: Optional[int] = None
    deployed: bool = False
    status: Literal["active", "paused", "terminated"] = "active"
    priority: Optional[int] = None
    failure_strategies: Optional[List[str]] = None
    collaboration_mode: Literal["single", "team", "parallel", "sequential"] = "single"
    available_for_deployment: bool = True
    deployment_code: Optional[str] = None

class AgentFunctionInner(BaseModel):
    name: str

class AgentFunction(BaseModel):
    function: AgentFunctionInner

class AgentCreate(AgentBase):
    functions: List[AgentFunction]

class AgentUpdate(BaseModel):
    name: Optional[str] = None
    agent_type: Optional[
        Literal[
            "router",
            "web_search",
            "data",
            "troubleshooting",
            "api",
            "weather",
            "toshiba",
        ]
    ] = None
    description: Optional[str] = None
    parent_agent_id: Optional[uuid.UUID] = None
    system_prompt_id: Optional[uuid.UUID] = None
    persona: Optional[str] = None
    functions: Optional[List[AgentFunction]] = None
    routing_options: Optional[Dict[str, str]] = None
    short_term_memory: Optional[bool] = None
    long_term_memory: Optional[bool] = None
    reasoning: Optional[bool] = None
    input_type: Optional[List[Literal["text", "voice", "image"]]] = None
    output_type: Optional[List[Literal["text", "voice", "image"]]] = None
    response_type: Optional[Literal["json", "yaml", "markdown", "HTML", "None"]] = None
    max_retries: Optional[int] = None
    timeout: Optional[int] = None
    deployed: Optional[bool] = None
    status: Optional[Literal["active", "paused", "terminated"]] = None
    priority: Optional[int] = None
    failure_strategies: Optional[List[str]] = None
    collaboration_mode: Optional[Literal["single", "team", "parallel", "sequential"]] = None
    available_for_deployment: Optional[bool] = None
    deployment_code: Optional[str] = None

class AgentInDB(AgentBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    agent_id: uuid.UUID
    session_id: Optional[str] = None
    last_active: Optional[datetime] = None
    functions: List[ChatCompletionToolParam]

class AgentResponse(AgentInDB):
    system_prompt: PromptResponse
    functions: List[ChatCompletionToolParam]

File: elevaite/apps/command_agent_version1/app/lib/interfaces/agents.ts
import { AgentType } from "../../components/type";
import { ChatCompletionToolParam, AgentFunction } from "./common";

export interface PromptResponse {
  prompt_label: string;
  prompt: string;
  unique_label: string;
  app_name: string;
  version: string;
  ai_model_provider: string;
  ai_model_name: string;
  tags?: string[] | null;
  hyper_parameters?: Record<string, string> | null;
  variables?: Record<string, string> | null;
  id: number;
  pid: string;
  sha_hash: string;
  is_deployed: boolean;
  created_time: string;
  deployed_time?: string | null;
  last_deployed?: string | null;
}

export interface AgentResponse {
  name: string;
  agent_type?: AgentType | null;
  description?: string | null;
  parent_agent_id?: string | null;
  system_prompt_id: string;
  persona?: string | null;
  functions: ChatCompletionToolParam[];
  routing_options: Record<string, string>;
  short_term_memory: boolean;
  long_term_memory: boolean;
  reasoning: boolean;
  input_type: ("text" | "voice" | "image")[];
  output_type: ("text" | "voice" | "image")[];
  response_type: "json" | "yaml" | "markdown" | "HTML" | "None";
  max_retries: number;
  timeout?: number | null;
  deployed: boolean;
  status: "active" | "paused" | "terminated";
  priority?: number | null;
  failure_strategies?: string[] | null;
  collaboration_mode: "single" | "team" | "parallel" | "sequential";
  available_for_deployment: boolean;
  deployment_code?: string | null;
  id: number;
  agent_id: string;
  session_id?: string | null;
  last_active?: string | null;
  system_prompt: PromptResponse;
}

export interface AgentConfigData {
  agentName: string;
  deploymentType: string;
  modelProvider: string;
  model: string;
  outputFormat: string;
  selectedTools: ChatCompletionToolParam[];
}

export interface AgentCreate {
  name: string;
  agent_type?: AgentType | null;
  description?: string | null;
  parent_agent_id?: string | null;
  system_prompt_id: string;
  persona?: string | null;
  routing_options: Record<string, string>;
  short_term_memory: boolean;
  long_term_memory: boolean;
  reasoning: boolean;
  input_type: ("text" | "voice" | "image")[];
  output_type: ("text" | "voice" | "image")[];
  response_type: "json" | "yaml" | "markdown" | "HTML" | "None";
  max_retries: number;
  timeout?: number | null;
  deployed: boolean;
  status: "active" | "paused" | "terminated";
  priority?: number | null;
  failure_strategies?: string[] | null;
  collaboration_mode: "single" | "team" | "parallel" | "sequential";
  available_for_deployment: boolean;
  deployment_code?: string | null;
  functions: AgentFunction[];
}

export interface AgentUpdate {
  name?: string;
  agent_type?: AgentType | null;
  description?: string | null;
  parent_agent_id?: string | null;
  system_prompt_id?: string;
  persona?: string | null;
  functions?: AgentFunction[];
  routing_options?: Record<string, string>;
  short_term_memory?: boolean;
  long_term_memory?: boolean;
  reasoning?: boolean;
  input_type?: ("text" | "voice" | "image")[];
  output_type?: ("text" | "voice" | "image")[];
  response_type?: "json" | "yaml" | "markdown" | "HTML" | "None";
  max_retries?: number;
  timeout?: number | null;
  deployed?: boolean;
  status?: "active" | "paused" | "terminated";
  priority?: number | null;
  failure_strategies?: string[] | null;
  collaboration_mode?: "single" | "team" | "parallel" | "sequential";
  available_for_deployment?: boolean;
  deployment_code?: string | null;
}

export interface AgentNodeData {
  id: string;
  shortId?: string;
  type: AgentType;
  name: string;
  prompt?: string;
  tools?: ChatCompletionToolParam[];
  config?: AgentConfigData;
  onDelete: (id: string) => void;
  onConfigure: (id: string) => void;
  tags?: string[];
  agent: AgentResponse;
  description?: string;
}

export interface AgentExecutionRequest {
  query: string;
  session_id?: string;
  user_id?: string;
  chat_history?: Array<Record<string, any>>;
  enable_analytics?: boolean;
}

export interface AgentExecutionResponse {
  status: string;
  response: string;
  agent_id: string;
  execution_time?: number;
  timestamp: string;
}

export interface AgentStreamExecutionRequest {
  query: string;
  chat_history?: Array<Record<string, any>>;
}
